### PR TITLE
feat(fiat): Adds migrator to add execute permissions to applications.

### PR DESCRIPTION
**Pull request by dibyom**
Execute is added to all groups that currently have READ or WRITE permissions on an application. This is because currently anyone with READ permission can execute a pipeline.

Checks enforcing that users need EXECUTE to trigger a pipeline is coming in a follow up PR. 

Part of spinnaker/spinnaker#3554

### CODE REVIEW

File: front50-migrations/src/main/java/com/netflix/spinnaker/front50/migrations/ApplicationExecutePermissionMigration.java [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+  private static final LocalDate VALID_UNTIL = LocalDate.of(2020, Month.APRIL, 1);
+
+  private ApplicationPermissionDAO applicationPermissionDAO;
+
+  @Setter
+  private Clock clock = Clock.systemDefaultZone();
+
+  @Autowired
+  public ApplicationExecutePermissionMigration(
+      ApplicationPermissionDAO applicationPermissionDAO) {
```
Comment (ezimanyi) [inline front50-migrations/src/main/java/com/netflix/spinnaker/front50/migrations/ApplicationExecutePermissionMigration.java]: nit: indentation is a bit off here

File: front50-migrations/src/main/java/com/netflix/spinnaker/front50/migrations/ApplicationExecutePermissionMigration.java [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+import java.time.LocalDate;
+import java.time.Month;
+import java.util.ArrayList;
+
+@Component
+@Slf4j
+public class ApplicationExecutePermissionMigration implements Migration {
+
+  // Only valid until May 1st, 2020
+  private static final LocalDate VALID_UNTIL = LocalDate.of(2020, Month.APRIL, 1);
```
Comment (ezimanyi) [inline front50-migrations/src/main/java/com/netflix/spinnaker/front50/migrations/ApplicationExecutePermissionMigration.java]: There's a mismatch between the comment and the code (April vs. May)

File: front50-migrations/src/main/java/com/netflix/spinnaker/front50/migrations/ApplicationExecutePermissionMigration.java [hunk 3]
```diff
@@ -0,0 +1,68 @@
+package com.netflix.spinnaker.front50.migrations;
```
Comment (ezimanyi) [inline front50-migrations/src/main/java/com/netflix/spinnaker/front50/migrations/ApplicationExecutePermissionMigration.java]: Missing header (both files)

**Comment by ajordens** [comment]
What are your thoughts on handing the lack of explicit `EXECUTE` when someone is setting permissions on an application ... the migrator would _eventually_ detect it but not immediately.

**Comment by ajordens** [comment]
Good point about the fallbacks ... 

It seems plausible that we might be able to get away without a migration, as well as configuration point to determine how `EXECUTE` should be interpreted? The later may satisfy a desire of ours to move from `READ` == `EXECUTE` to `WRITE` == `EXECUTE`.

Could be simpler without much of a hit complexity wise in `FiatPermissionEvaluator`?

**Comment by jtk54** [comment]
Code looks good to me, but +1 to ajordens suggestion to explore whether we can get out of jail free without a migrator.

**Comment by dibyom** [comment]
@ezimanyi I like the idea. I'll just add the fallback check in FiatPermissionEvaluator. Defaults to anyone with read has execute but the option to change to write implies execute for those that want it. I don't think there would be any performance hit.

**Pull request closed without merging**